### ISSUE TITLE
Tab registration missing for extensions in Preview page

### ISSUE THREAD

**Issue by provisionally** [jqssun/android-helium-browser#6]
Putting https://github.com/imputnet/helium/issues/105#issuecomment-4539532376 in its proper place.

### Bugs
- [ ] uBlock Origin, not lite, does not work in the long-press `Preview page` pseudo tabs or in the tabs that open within apps when Helium is set as the device's default browser
- [x] Disabling `Show zoom option in main menu` in Accessibility fails and is immediately reenabled once you exit Accessibility
- [ ] When `Address bar` is set to `Bottom`, the url preview and suggestions list, as well as the omnibox tools, remain at the top
- [ ] The 3-dot main menu action buttons, that is reload and the other 4, also remain at the top despite the address bar set to the bottom making it hard to reach and it sometimes requires a scroll to reach since Helium has much more menu items

### Requests
- [ ] Please fix the Android Chromium regression that made the omnibox wipe every time it gets focus, forcing an unnecessary use of that pencil icon just to edit an url, no matter how minor
- [ ] View Source button next to Developer tools
- [x] An option to remove the Extensions menu icon from the address bar as it takes precious space after pinning two extensions

No idea if it is helpful as an implementation reference but the last two Bugs and first Request are fixed in Cromite on Android (github.com/uazo/cromite). They are the only things I really miss coming from there.

**Comment by provisionally** [comment]
### Bug
- [ ] The new tab page, shown especially when starting up Helium, still has what should be the bottom bar, with the extension and every thing else, at the top despite Address bar set to Bottom

### Request
- [ ] Incognito by default, something like `Add an always-incognito mode` in github.com/uazo/cromite/blob/master/docs/PATCHES.md, though that seems to be a bit buggy and will likely need exceptions for opening extensions in Helium that require normal mode

With respect to the very first request, the relevant flag is #enable-search-ready-omnibox. Disabling that used to fix the issue but Chrome removed it. Cromite reintroduced it with `Restore Search Ready Omnibox flag` in github.com/uazo/cromite/blob/master/docs/PATCHES.md.

**Comment by provisionally** [comment]
The bugs were observed in at least v149.0.7827.22 and still occur in the latest v149.0.7827.48.

**Comment by Brocman** [comment]
"When Address bar is set to Bottom, the url preview and suggestions list, as well as the omnibox tools, remain at the top"

<img width="1264" height="2780" alt="Image" src="https://github.com/user-attachments/assets/9495651f-b774-498a-b8db-7026812d8cef" />

Enabled Force bottom for
focused omnibox

**Comment by provisionally** [comment]
That does not do anything, neither here nor in any other Chromium fork tested. It is also disabled on Cromite which has everything in the right place.

**Comment by Brocman** [comment]
You can send two images or screenshots — one showing how it currently looks for you, and one showing how you want it to be.

**Comment by Brocman** [comment]
<img width="1264" height="2780" alt="Image" src="https://github.com/user-attachments/assets/f22cd350-4015-4517-8cda-3888e5018c18" />

<img width="1264" height="2780" alt="Image" src="https://github.com/user-attachments/assets/244fbdd6-0794-424a-bd13-f4f8df501d55" />

**Comment by provisionally** [comment]
Already covered that in the request. Cromite gets it right, where it looks like that screenshot.

**Comment by Brocman** [comment]
If you want it to look like my screenshots, enable the option "Enabled Reverse list and force bottom for focused omnibox"
chrome://flags/#android-bottom-toolbar-v2

**Comment by provisionally** [comment]
I already said that does not do anything.

**Comment by Brocman** [comment]
Then I don't understand what you want. If it's about where the search bar and suggestions are located when you tap it, you can change the behavior by doing what I described. This works in Chrome and Helium. If you're talking about other interface elements, like the browser's default home page or tabs, the controls are located at the top there.

**Comment by provisionally** [comment]
So `Enabled Force bottom for focused omnibox` and `Enabled Reverse list and force bottom for focused omnibox` are different. The first you suggested does nothing in every Chromium variant tested. The second does seem to fix the problem though it is strange it is meaningless in Cromite.

**Comment by stormcats** [comment]
Does the toolbar not disappear when scrolling? I got my toolbar on the bottom. Top or bottom doesn't disappear. Another thing, when holding the back button on android you can see the previous web pages you've visited on the current tab. Is that feature not there anymore?

**Comment by provisionally** [comment]
@jqssun Is the first bug even fixable? It has been a month now and with no updates to that effect, and since even Cromite has the same issue, I have been wondering if I may be waiting in vain.

**Comment by jqssun** [comment]
**View page source** should now be available in the context menu (consistent with ALOS). You can activate it by long pressing any webpage :)

**Comment by jqssun** [comment]
Can't seem to repro the Custom Tab bit of the issue. uBO (MV2) is working fine when started with

```shell
am start -a android.intent.action.VIEW --es android.support.customtabs.extra.SESSION "1" -d "https://google.com/"
```

**Comment by jqssun** [comment]
Will be tracking the remaining bottom bar related issues in #16 going forward

**Comment by jqssun** [comment]
Feature requests are accepted and will be tracked in #4

**Comment by provisionally** [comment]
> Can't seem to repro the Custom Tab bit of the issue.

@jqssun Seems like it, I never could test it since it is app-dependent, but `Preview page` is definitely broken. It is my single most used feature since its introduction in Brave and I have been waiting 7 months for a fix in Cromite, and a month here.

Considering it only affects MV2 extensions, which Google is desperate to kill, and in a place that seems mobile-exclusive, where they do not officially support extensions, not only will it be painful to have to wait for Google for salvation because that might very well take 5 years, but it also seems fruitless.

**Comment by jqssun** [comment]
@provisionally 
Interesting, MV2 extensions should be working in **Preview page** on the latest version. A simple test case was to only enable uBO (MV2) and check whether the ad block works. It does work in the window created by the preview on my end - what is your use case?

**Comment by provisionally** [comment]
Definitely not working here. 

1. On a Pixel, go to some site that is known to have ads, I guess something like the-sun.com or maybe even 9to5google.com
2. Load an article on the site and observe everything loading ad-free, with no giant gaps
3. Return to the homepage of the site and then long-press on the link of the article from 2 and then select `Preview page`
4. Observe there are either actual ads, giant gaps, or dark blocks with `Advertisement` written in them

One thing to note is that it seems Vanadium includes some basic ad-block stuff itself, which Helium inherits, so within the `Preview page`, you might not see ad content but 4 definitely happens and in some cases, you even get `Adblock detected`. That is not from uBlock working.

**Comment by provisionally** [comment]
> View page source should now be available in the context menu

I do not mind this but can you add it to the main menu too. The problem with the context menu design is that it requires having some free space on the page to long-press that is not a link or similar which might not be possible or require scrolling all the way to the top.

**Comment by provisionally** [comment]
Clearer way to confirm extensions are definitely not working in `Preview page` is to use a ViolentMonkey script for an article, like maybe change the font size to something ridiculous. In a normal tab, the script executes the effect on the page. In `Preview page`, nothing happens.

**Comment by jqssun** [comment]
@provisionally Hmm that's interesting... ViolentMonkey works in preview on my side; for context I tested it with [this script](https://greasyfork.org/en/scripts/1-greasemonkey-tampermonkey-violentmonkey-test-style) (so it just turns everything red).

I wonder if it's because few APIs aren't available in previews. RE the page source button, the platform native implementation is in the context menu (the way Chromium intended), but I can make that button appear elsewhere too but it will be missing an icon by default.

**Comment by provisionally** [comment]
> ViolentMonkey works in preview on my side; for context I tested it with this script (so it just turns everything red)

@jqssun That works for me too. Very strange. Things like `alert` do not work from within the same script though, and I wonder about `console.log` too but no idea how to even test that.

uBlock Origin still does not work even after updating and force-stopping Helium. I am having a hard time coming up with a site to show the clear contrast and Vanadium's very basic filters is complicating matters but try this:

1. Clear all browsing data 
2. In uBlock's dashboard under `Filter lists`, enable all under `Cookie notices` and `EasyList - Annoyances`, and `uBlock filters - Annoyances` and `AdGuard - Mobile App Banners`
3. In an incognito tab, go to reddit.com/r/all or reddit.com/r/popular, specifically reddit.com, not old.reddit
4. Everything should work as expected and you should be able to navigate freely, no ads
5. Long-press `Preview page` any thread, the page should now be unscrollable and then become almost entirely covered by a cookie notice
6. Close the preview and click on the same thread in a normal tab, no notice and perfectly scrollable

**Comment by provisionally** [comment]
@jqssun Can you reopen it now? It looks weird closed.

**Comment by provisionally** [comment]
Possibly clearer example that demonstrates the issue:

1. Follow all the steps from earlier, but do not go to reddit
2. In an incognito tab, google search for something that brings up an `AI Overview` and refresh a few times to confirm it shows consistently
3. Go to `My filters` in uBlock's dashboard and erase or comment out anything there
4. In a new line, write in `google.*##div.MjjYud:has(> div#eKIzJc > .YzCcne)` and save
5. Return to the search and refresh, observe the `AI Overview` is gone, refresh a few times to confirm
6. Long-press `Preview page` the `All` in the tab strip underneath the Google results page searchbox
7. Observe the `AI Overview` has returned like the filters do not exist
8. Close the preview and refresh the search again, `AI Overview` remains gone

This might be a long shot but, can you please help, @gorhill?

**Comment by provisionally** [comment]
@ArseniyW Does it really not work in those tabs? It thought the same at first but later on, it seems it just does not show the pinned extension buttons, which is admittedly a big deal for something like uBlock Origin.

**Comment by ArseniyW** [comment]
Yes, that's right, but I often use extensions like "Linguist", and I can't enable this extension in the preview

**Comment by provisionally** [comment]
> and I can't enable this extension in the preview

@ArseniyW That is `Preview page` only or also in the custom tabs?

**Comment by ArseniyW** [comment]
Preview page only

**Comment by provisionally** [comment]
Ok, same here. That and the crashes are the primary things holding things back for me so far.

**Comment by provisionally** [comment]
@NF075 Put that in its own issue, not here.

@jqssun Any luck with this bug?

**Comment by provisionally** [comment]
@jqssun The `Bottom Toolbar V2` flag has been removed so that issue is relevant again.

**Comment by provisionally** [comment]
@jqssun Any luck?
